From 3e0fa14847b089291f4ca990d43029c1cd77a232 Mon Sep 17 00:00:00 2001
From: Vijay Bellur <vbellur@redhat.com>
Date: Thu, 31 May 2018 17:11:01 -0700
Subject: [PATCH 294/305] features/shard: Fix missing unlock in
 shard_fsync_shards_cbk()

updates: bz#789278

> Upstream: https://review.gluster.org/20118
> BUG: 789278
> Change-Id: I745a98e957cf3c6ba69247fcf6b58dd05cf59c3c

Change-Id: I745a98e957cf3c6ba69247fcf6b58dd05cf59c3c
BUG: 1493085
Signed-off-by: Vijay Bellur <vbellur@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/140462
Tested-by: Krutika Dhananjay <kdhananj@redhat.com>
Tested-by: RHGS Build Bot <nigelb@redhat.com>
Reviewed-by: Sunil Kumar Heggodu Gopala Acharya <sheggodu@redhat.com>
---
 xlators/features/shard/src/shard.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xlators/features/shard/src/shard.c b/xlators/features/shard/src/shard.c
index 29989d3..d67cdf4 100644
--- a/xlators/features/shard/src/shard.c
+++ b/xlators/features/shard/src/shard.c
@@ -4504,6 +4504,7 @@ shard_fsync_shards_cbk (call_frame_t *frame, void *cookie, xlator_t *this,
                 if (op_ret < 0) {
                         local->op_ret = op_ret;
                         local->op_errno = op_errno;
+                        UNLOCK (&frame->lock);
                         goto out;
                 }
                 shard_inode_ctx_set (local->fd->inode, this, postbuf, 0,
-- 
1.8.3.1

